service/route_handlers.go

```diff
@@ -400,3 +402,33 @@ func DoSamlLogout(w http.ResponseWriter, r *http.Request) {
 		log.Info("No Logout URL - Saml/Shibboleth provider is not configured")
 	}
 }
+
+// TestLogin is a test API to check login with code before saving settings to db
+func TestLogin(w http.ResponseWriter, r *http.Request) {
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("TestLogin failed with error: %v", err)
+		ReturnHTTPError(w, r, http.StatusBadRequest, "Bad Request, Please check the request content")
+		return
+	}
+	var testAuthConfig model.TestAuthConfig
+
+	err = json.Unmarshal(bytes, &testAuthConfig)
+	if err != nil {
+		log.Errorf("TestLogin unmarshal failed with error: %v", err)
+		ReturnHTTPError(w, r, http.StatusBadRequest, "Bad Request, Please check the request content")
+		return
+	}
+
+	if testAuthConfig.AuthConfig.Provider == "" {
+		log.Errorf("UpdateConfig: Provider is a required field")
+		ReturnHTTPError(w, r, http.StatusBadRequest, "Bad Request, Please check the request content, Provider is a required field")
+		return
+	}
+
+	err = server.GetProvider(testAuthConfig)
+	if err != nil {
+		log.Errorf("TestLogin GetProvider failed with error: %v", err)
+		ReturnHTTPError(w, r, http.StatusBadRequest, "Bad Request, Please check the request content")
```

Comment:
Should we be retuning 401 since the login failed? This returns 404. Please check with @vincent99 if returning a 401 will be a problem for the UI client